tests/functions/test_pytorch_model.py

```diff
@@ -286,8 +293,40 @@ def test_add_mean_shift_to_first_layer(load_mnist_model):
         ),
     ],
 )
-def test_huggingface_classifier_predict(hf_model, data, softmax, model_kwargs, expected):
-    model = PyTorchModel(model=hf_model, softmax=softmax, model_predict_kwargs=model_kwargs)
+def test_huggingface_classifier_predict(
+    hf_model, data, softmax, model_kwargs, expected
+):
+    model = PyTorchModel(
+        model=hf_model, softmax=softmax, model_predict_kwargs=model_kwargs
+    )
     with expected:
         out = model.predict(x=data)
         assert np.allclose(out, expected.enter_result), "Test failed."
+
+
+@pytest.mark.pytorch_model
+@pytest.mark.parametrize(
+    "transformers_installed,base_class,expected",
+    [
+        (True, PreTrainedModel, nullcontext(np.array([[0.1, 0.9]], dtype=np.float32))),
+        (False, None, pytest.raises(ValueError)),
+    ],
+)
+def test_predict_transformers_installed(
+    mocker, transformers_installed, base_class, expected
+):
+    mocker.patch("importlib.util.find_spec", return_value=transformers_installed)
+    from quantus.helpers.model import pytorch_model
+
+    reload(pytorch_model)
+    # Mock the model's behavior
+    model_instance = PyTorchModel(model=mocker.MagicMock(spec=base_class))
```

Comment:
Hi @abarbosa94, 

could you please make sure the original state is restored after this function exits? This can be done using context-manager fixture, e.g., https://github.com/understandable-machine-intelligence-lab/Quantus/blob/main/tests/metrics/conftest.py